src/utilities.ts

```diff
@@ -94,38 +98,52 @@ export async function AddUserBlockHistory(
 	}
 
 	let response: MessageResponse | null = null;
-	for (let i = 0; i < 5; i++) {
-		const d = { action: AddToHistoryAction, data };
-		response = (await api.runtime.sendMessage(d)) as MessageResponse;
+	const maxAttempts = 5;
+	let attempt: number = 0;
+	for (;;) {
```

Comment:
Can't you do:
```suggestion
    const maxAttempts = 5;
    for (let attempt = 0; attempt < maxAttempts; attempt++) {
```

Reply:
yeah I just didn't want to do the `attempt < maxAttempts` check twice back-to-back, it just bothers me, but this would be perfectly valid

Reply:
I think if you have the comparison in the for statement you could just put the delay in an else block?

Reply:
I'm not sure what you mean by this, there is a comparator in the for loop definition as well as within the loop itself, but if you define the loop as `;;` then there is only the one within the loop

Reply:
oh I looked at your code again, I think I know what you mean. I don't want to do it like that because the wait will occur on the final loop that way.

if the function were to error, it would loop 5 times, and on the last time (attempt == 4) it will wait unnecessarily, as there will not be another attempt, for 16 seconds.

Reply:
I guess then it would probably make more sense to turn the whole thing into a do ... while loop right? The condition gets checked after execution so you would exit after 4 loop throughs.
```ts
do {
   // all of the stuff
} while (attempt < maxAttempts)
```